homeassistant/components/awair/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for Awair."""
+
+from typing import Union
+
+from python_awair import Awair
+from python_awair.exceptions import AuthError, AwairError
+import voluptuous as vol
+
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, ConfigFlow
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, LOGGER  # pylint: disable=unused-import
+
+
+class AwairFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Awair."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    async def async_step_import(self, conf: dict):
+        """Import a configuration from config.yaml."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_setup")
+
+        if not conf[CONF_ACCESS_TOKEN]:
+            return self.async_abort(reason="auth")
+
+        await self._abort_if_configured(conf[CONF_ACCESS_TOKEN])
+
+        user, errors = await self._check_access_token(conf[CONF_ACCESS_TOKEN])
+        if errors is not None:
+            return self.async_abort(reason="auth")
+
+        return self.async_create_entry(
+            title=f"{user.email} ({user.user_id})",
+            data={CONF_ACCESS_TOKEN: conf[CONF_ACCESS_TOKEN]},
+        )
+
+    async def async_step_user(self, user_input: Union[dict, None] = None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            user, errors = await self._check_access_token(user_input[CONF_ACCESS_TOKEN])
+
+            if not errors:
+                await self._abort_if_configured(user_input[CONF_ACCESS_TOKEN])
+                title = f"{user.email} ({user.user_id})"
+                return self.async_create_entry(title=title, data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_ACCESS_TOKEN): str}),
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, user_input: Union[dict, None] = None):
+        """Handle re-auth if token invalid."""
+        errors = {}
+
+        if user_input is not None:
+            if "config_entry" in user_input:
```

Comment:
We can only send the config entry data as input to this step, as it has the same format as the form input.

Reply:
Technically ... this does seem to work as-written, but I'm happy to change it to be a little more straightforward if you'd like.

Related to your comment re: "should we really use the token as the unique ID" - the only reason I sent the config entry over in the first place is that an earlier iteration of this was using the entry ID in a `self.async_update_entry` call, but it didn't survive refactoring.

That raises the question: if the unique ID is derived from user input, and we want to *update* the entry for reauth, what's the recommended way to get the original entry to update? (I'm not sure if I'm making myself clear, but this was a legit problem I ran into - I couldn't call `update` on the original entry since the user input (which derives the unique id) would change when it was corrected).

Reply:
I've switched to using a user-provided email address as the unique ID, and this simplifies things quite a bit (in my opinion, anyways). It's an otherwise unneeded piece of info from the user but I think it's a fine trade-off. I'm passing the unique_id of the currently-existing config entry via context, and that seems to work just fine overall.